hieradata/clients/azure.ci.jenkins.io.yaml

```
@@ -20,7 +20,7 @@ profile::buildmaster::groovy_d_lock_down_jenkins: 'present'
 profile::buildmaster::groovy_d_pipeline_configuration: 'present'
 profile::buildmaster::groovy_d_set_up_git: 'present'
 profile::buildmaster::groovy_d_terraform_credentials: 'present'
-profile::buildmaster::java_opts: "-server -Xlog:gc+heap*=debug,ref*=debug,ergo*=trace,age*=trace:file=/var/jenkins_home/gc-%t.log::filecount=5,filesize=20M -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:+UnlockDiagnosticVMOptions -Xms16g -Xmx16g -Duser.home=/var/jenkins_home -Djenkins.install.runSetupWizard=false -Djenkins.model.Jenkins.slaveAgentPort=50000 -Dhudson.model.WorkspaceCleanupThread.retainForDays=2" # Java11 specific configuration
+profile::buildmaster::java_opts: "-server -Xlog:gc+heap*=debug,ref*=debug,ergo*=trace,age*=trace:file=/var/jenkins_home/gc/gc.log::filecount=5,filesize=40M -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:+UnlockDiagnosticVMOptions -Xms16g -Xmx16g -Duser.home=/var/jenkins_home -Djenkins.install.runSetupWizard=false -Djenkins.model.Jenkins.slaveAgentPort=50000 -Dhudson.model.WorkspaceCleanupThread.retainForDays=2" # Java11 specific configuration
```

```suggestion
profile::buildmaster::java_opts: "-server -Xlog:gc*=info,ref*=debug,ergo*=trace,age*=trace:file=/var/jenkins_home/gc/gc.log::filecount=5,filesize=40M -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:+UnlockDiagnosticVMOptions -Xms16g -Xmx16g -Duser.home=/var/jenkins_home -Djenkins.install.runSetupWizard=false -Djenkins.model.Jenkins.slaveAgentPort=50000 -Dhudson.model.WorkspaceCleanupThread.retainForDays=2" # Java11 specific configuration
```

@ryansmith303 proposes that we remove the debug GC logs and thus use these settings. cc @olblak @daniel-beck @batmat @rsandell @oleg-nenashev

---

The continued presence of this after someone identified it as a problem in the call confused me anyway.

---

yes it was never fixed. 

Originally we had (ref: https://github.com/jenkins-infra/jenkins-infra/blob/d6e9e1c8ab46b0c89816a604f57a5ed26a724437/hieradata/clients/azure.ci.jenkins.io.yaml#L23 ):

```
-Xlog:gc+heap*=debug,ref*=debug,ergo*=trace,age*=trace:file=/var/jenkins_home/gc-%t.log::filecount=5,filesize=20M
``` 

We now proposes

```
-Xlog:gc*=info,ref*=debug,ergo*=trace,age*=trace:file=/var/jenkins_home/gc/gc.log::filecount=5,filesize=40M
```

* Info level by default
* No timestamp in the filename
* Logs not directly under JENKINS_HOME (but @olblak is verifying if they could be outside of JENKINS_HOME instead in a subfolder)
* Larger files (40Mb instead of 20Mb)

---

> if they could be outside of JENKINS_HOME instead in a subfolder

What's the benefit of doing that?

---

I think it is to avid to have them grabbed by default with by support-core but my understanding is that in any case it doesn't work with J11 (@dohbedoh ?)
Also potentially to not archive/backup them but now if we don't use a timestamp it's maybe not an issue...

---

> I think it is to avid to have them grabbed by default with by support-core

Already accomplished through the subdirectory.

> Also potentially to not archive/backup them

5GB so far don't even matter among all the other nonsense.